sos/plugins/crash_report.py

```diff
@@ -0,0 +1,249 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+import os
+import sys
+import time
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+from sos.utilities import sos_get_command_output, find
+from datetime import datetime
+
+
+class CrashLog(Plugin):
+    """Crash kernel report
+    """
+    plugin_name = "crash_report"
+    profiles = ('system', 'debug')
+
+    dbg_vmlinux_path = '/usr/lib/debug/'
+    vmcore_stnd_path = '/var/crash'
+
+    dump_file_prefix = ''
+    crashed_kernel_name = ''
+    vmlinux_path = ''
+    message_to_log = ''
+    vmcore_file_path = ''
+    crash_script = ""
+    # crash result output file
+    crashed_output_file = '/var/log/os_crash_report.txt'
+
+    # list of crash command
+    crash_cmd_list = [
+                    ("sys", "\"SYSTEM INFORMATION:\""),
+                    ("mach", "\"MACHINE SPECIFIC DATA:\""),
+                    ("mach", "\"MACHINE SPECIFIC DATA:\""),
+                    ("bt", "\"STACK TRACE OF CURRENT CONTEXT:\""),
+                    ("bt -a", "\"STACK TRACES OF ACTIVE TASKS:\""),
+                    ("kmem -i", "\"MEMORY USAGE:\""),
+                    ("kmem -s", "\"KMALLOC SLAB DATA:\""),
+                    ("mod", "\"MODULES:\""),
+                    ("ps", "\"PROCESS STATUS:\""),
+                    ("log", "\"SYSTEM MESSAGE BUFFER:\""),
+                    ("files", "\"OPEN FILES OF CURRENT CONTEXT:\""),
+                    ("dev -p", "\"PCI DEVICE DATA:\""),
+                    ("runq", "\"RUN QUEUE TASK:\""),
+                    ("mach -o", "\"OPALMSG LOG:\""),
```

Comment:
Hi Robb,
Yes, mach -o is supported only on PPC64 systems.

I prefer logging error in case of other architecture rather checking help command (check will increase complexity of this unnecessarily ).

We should keep "mach -o" part of this plugin as it is needed on PPC64 architecture and even on PPC64 sos is an important tool to collect data.